tests/federation/test_federation_sender.py

```diff
@@ -473,13 +481,15 @@ def test_upload_signatures(self):
             self.assertEqual(edu["edu_type"], EduTypes.DEVICE_LIST_UPDATE)
             c = edu["content"]
             if stream_id is not None:
-                self.assertEqual(c["prev_id"], [stream_id])
+                # type-ignore: mypy marks the next line as unreachable; it does not seem
+                # to recognise the assignment to stream_id within the body of this loop
```

Comment:
I think if you add an explicit `stream_id: Optional[str] = None` you can remove the ignore?

Reply:
Odd, I could have sworn I did that somewhere earlier in the function... seemingly not!

Reply:
Ahh, I'm thinking of line 568 below.

Reply:
This is 9d770bf37e